frame/composable-traits/src/dex.rs

```diff
@@ -12,16 +12,37 @@ pub trait CurveAmm {
 	type Balance;
 	/// The user account identifier type for the runtime
 	type AccountId;
-	/// Type that represents index type of token in the pool passed from the outside as an extrinsic
-	/// argument.
-	type PoolTokenIndex;
 
 	/// Type that represents pool id
 	type PoolId;
 
 	/// Current number of pools (also ID for the next created pool)
 	fn pool_count() -> Self::PoolId;
 
+	/// Get pure exchange value for given units of given asset. (Note this does not include fees.)
+	fn get_exchange_value(
+		pool_id: Self::PoolId,
+		asset_id: Self::AssetId,
+		amount: Self::Balance,
+	) -> Result<Self::Balance, DispatchError>;
+
+	/// Buy given units of given asset from the pool.
+	/// In buy user does not know how much assets he/she has to exchange to get desired amount.
+	fn buy(
+		who: &Self::AccountId,
+		pool_id: Self::PoolId,
+		asset_id: Self::AssetId,
+		amount: Self::Balance,
+	) -> Result<Self::Balance, DispatchError>;
+
+	/// Sell given units of given asset to the pool.
+	fn sell(
+		who: &Self::AccountId,
+		pool_id: Self::PoolId,
+		asset_id: Self::AssetId,
+		amount: Self::Balance,
```

Comment:
please comment what amount means. still not clear from comments what is difference of sell vs buy.